components/core/ConfirmationModal.js

```diff
@@ -0,0 +1,137 @@
+import * as React from "react";
+import * as Constants from "~/common/constants";
+import * as Validations from "~/common/validations";
+
+import { css } from "@emotion/react";
+import { useState } from "react";
+
+import { ButtonPrimaryFull, ButtonDeleteFull, ButtonDeleteDisabledFull, ButtonCancelFull } from "~/components/system/components/Buttons.js";
+import { Input } from "~/components/system/components/Input.js";
+
+const STYLES_TRANSPARENT_BG = css `
+  background-color: ${Constants.system.bgBlurGrayBlack};
+  width: 100%;
+  height: 100vh;
+  position: fixed;
+  left: 0;
+  top: 0;
+  z-index: 999999;
+`;
+
+const STYLES_MAIN_MODAL = css `
+  width: 380px;
+  height: auto;
+  background-color: ${Constants.system.white};
+  position: fixed;
+  left: 50%;
+  top: 50%;
+  transform: translate(-50%, -50%);
+  border-radius: 8px;
+  padding: 24px;
+  text-align: left;
+`;
+
+const STYLES_HEADER = css `
+  color: ${Constants.system.black};
+  font-size: ${Constants.typescale.lvl1};
+  font-family: ${Constants.font.semiBold};
+`;
+
+const STYLES_SUB_HEADER = css `
+  color: ${Constants.system.textGray};
+  font-size: ${Constants.typescale.lvl0};
+  margin-top: 16px;
+  font-family: ${Constants.font.text};
+`;
+
+const STYLES_INPUT_HEADER = css `
+  color: ${Constants.system.black};
+  font-size: ${Constants.typescale.lvlN1};
+  font-weight: bold;
+  margin-top: 24px;
+  margin-bottom: 8px;
+`;
+
+export const DeleteModal = (props) => {
```

Comment:
To make this function more flexible, I think a better paradigm is to think of it as something where you pass in the header and subheader, and it renders those. Rather than storing all the headers and subheaders inside the component. Because imagine if we start using this a lot, we'd have to keep adding more and more to it for the various options.

Using this component should look something like: 
``` 
<DeleteModal 
    header={`Are you sure you want to delete the selected ${props.data.selected} files?`}
    subHeader={'These files will be deleted from all connected collections and your file library. You can’t undo this action.'}
/>
```

And for the specific case where you need to validate your username, you can just add a flag for that:
``` 
<DeleteModal 
    header={`Are you sure you want to delete your account?`}
    subHeader={'Type your username to continue'}
    inputValidation=true
    matchValue={this.props.viewer.username}
/>
```

Reply:
And on top of that, I don't think you need 3 separate modals for Confirm,Delete, and DeletewithInput. These can all be the same component, but with a flag like "type" that determines which style it is.

```
<ConfirmationModal
    type={"DELETE"} // I would recommend all caps for types. makes it so people don't have to think about capitalization
/>
```